### PR TITLE
Return nil explicitly when there is no error.

### LINKED ISSUE #536: Prefer to explicitly return nil when there is no error for code readability
A common pattern in this repo is:

```go
if err != nil {
  return ..., err
}
return ..., err
```

and we would like to change the last return to explicitly `return ..., nil` for Go readability.

### PR DESCRIPTION
**Pull request by nspragg**
Resolves #536. Explicitly return nil when there is no error. Hopefully, this should address outstanding instances of the following pattern: 

```go
if err != nil {
  return ..., err
}
return ..., err
```

### CODE REVIEW

File: github/github.go [hunk 1]
```diff
@@ -739,7 +739,7 @@ func (c *Client) RateLimit() (*Rate, *Response, error) {
 		return nil, nil, err
 	}
 
-	return limits.Core, resp, err
+	return limits.Core, resp, nil
```
Comment (shurcooL) [inline github/github.go]: Hmm, this one won't work, it changes behavior. But I think the problem lies above.

Currently, it checks if limits is nil, and always returns err. In the (possibly impossible) path where limits is not nil, but err is also not nil, this change would incorrectly return nil error.

However, I think you should change the checks above to be like this:

```Go
limits, resp, err := c.RateLimits()
if err != nil {
	return nil, nil, err
}
if limits == nil {
	return nil, nil, fmt.Errorf("RateLimits() returned nil error and nil limits, cannot extract Core rate limit")
}
```

Then it's fine to do `return limits.Core, resp, nil` here.

We could also leave this as is, but I think it's unfriendly API to return `nil, nil, nil`, so changing it to always return non-nil error on failure would be better.

Comment (nspragg) [inline github/github.go]: Well spotted. I agree with checking both limit and error in this case (better than returning nil, nil, nil)

Incidentally, no tests failed. Is there any motivation to increase the test coverage in the project in the future?

Comment (shurcooL) [inline github/github.go]: Having high but practical test coverage is always a goal, and I think we're doing pretty well there (look at all the _test.go files). This finding doesn't change that. This is a particular rare edge case that I wouldn't expect to be fully caught by unit tests.

Also note that this is a deprecated method, so it's not worth spending a lot of time/effort on.

File: github/github.go [hunk 2]
```diff
@@ -735,9 +735,12 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
-	if limits == nil {
+	if err != nil {
 		return nil, nil, err
 	}
+	if limits == nil {
+		return nil, nil, fmt.Errorf("RateLimits returned nil limits and error. Unable to extract Core rate limit")
```
Comment (shurcooL) [inline github/github.go]: I think we should return original `resp` instead of `nil` here. It's either going to be nil or some valid response. If there's an error, and a valid response, it's better to return that to the caller so they can look into why the error happened, rather than dropping the response information.

Here, and in the if check above.

Comment (shurcooL) [inline github/github.go]: About the error message, see https://github.com/golang/go/wiki/CodeReviewComments#error-strings.

I don't think you should separate it into multiple sentences and capitalize the letters. First letter happens to be capital because it refers to `RateLimits` the proper noun.

File: github/github.go [hunk 3]
```diff
@@ -735,11 +735,14 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
+	if err != nil {
+		return nil, resp, err
+	}
 	if limits == nil {
-		return nil, nil, err
+		return nil, resp, fmt.Errorf("RateLimits returned nil limits and error; unable to extract Core rate limit")
 	}
```
Comment (shurcooL) [inline github/github.go]: Nitpick, but I would remove this blank line now. It does not provide value at this point IMO.

Without the blank line separating them, it would be a little more visible that `limits` is checked for being non-`nil` before dereferencing `limits.Core`.

File: github/github.go [hunk 4]
```diff
@@ -735,11 +735,13 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
+	if err != nil {
+		return nil, resp, err
+	}
 	if limits == nil {
-		return nil, nil, err
+		return nil, resp, fmt.Errorf("RateLimits returned nil limits and error; unable to extract Core rate limit")
```
Comment (gmlewis) [inline github/github.go]: s/fmt.Errorf/errors.New/
